internal/azurelinux/azurelinux.go

```diff
@@ -0,0 +1,451 @@
+		return err
+	}
+	return nil
+}
+
+func (v *Version) parseSpecVersion() (version *goversion.GoVersion, revision int, err error) {
+	version, err = v.parseVersion()
+	if err != nil {
+		return nil, 0, err
+	}
+	revision, err = v.parseRelease()
+	if err != nil {
+		return nil, 0, err
+	}
+	return version, revision, nil
+}
+
+func (v *Version) regexpReplace(re *regexp.Regexp, replacement string) error {
+	if !re.Match(v.Spec) {
+		return fmt.Errorf("no match found in spec content")
+	}
+	v.Spec = re.ReplaceAll(v.Spec, []byte(replacement))
+	return nil
+}
+
+var specFileVersionRegex = regexp.MustCompile(`(Version: +)(.+)`)
+
+func (v *Version) parseVersion() (*goversion.GoVersion, error) {
+	matches := specFileVersionRegex.FindSubmatch(v.Spec)
+	if matches == nil {
+		return nil, fmt.Errorf("no Version declaration found in spec content")
+	}
+	return goversion.New(string(matches[2])), nil
+}
+
+func (v *Version) updateVersion(newVersion string) error {
+	return v.regexpReplace(specFileVersionRegex, "${1}"+escapeRegexReplacementValue(newVersion))
+}
+
+var specFileReleaseRegex = regexp.MustCompile(`(Release: +)(.+)(%\{\?dist\})`)
+
+func (v *Version) parseRelease() (int, error) {
+	matches := specFileReleaseRegex.FindSubmatch(v.Spec)
+	if matches == nil {
+		return 0, fmt.Errorf("no Release declaration found in spec content")
+	}
+	releaseInt, err := strconv.Atoi(string(matches[2]))
+	if err != nil {
+		return 0, fmt.Errorf("failed to parse old Release number: %v", err)
+	}
+	return releaseInt, nil
+}
+
+func (v *Version) updateRelease(newRelease string) error {
+	return v.regexpReplace(
+		specFileReleaseRegex,
+		"${1}"+escapeRegexReplacementValue(newRelease)+"${3}")
+}
+
+var specFileGoFilenameRegex = regexp.MustCompile(`(%global ms_go_filename +)(.+)`)
+
+func (v *Version) parseGoArchiveName() (string, error) {
+	matches := specFileGoFilenameRegex.FindSubmatch(v.Spec)
+	if matches == nil {
+		return "", fmt.Errorf("no Go archive filename declaration found in spec content")
+	}
+	return strings.TrimSpace(string(matches[2])), nil
+}
+
+func (v *Version) updateGoArchiveName(newArchiveName string) error {
+	return v.regexpReplace(
+		specFileGoFilenameRegex,
+		"${1}"+escapeRegexReplacementValue(newArchiveName))
+}
+
+var specFileRevisionRegex = regexp.MustCompile(`(%global ms_go_revision +)(.+)`)
+
+func (v *Version) updateGoRevision(newRevisionVersion string) error {
+	if !specFileRevisionRegex.Match(v.Spec) {
+		return fmt.Errorf("no Go revision version declaration found in spec content")
+	}
+	v.Spec = specFileRevisionRegex.ReplaceAll(
+		v.Spec,
+		[]byte("${1}"+escapeRegexReplacementValue(newRevisionVersion)),
+	)
+	return nil
+}
+
+func (v *Version) addChangelog(changelogDate time.Time, assets *buildassets.BuildAssets, newVersion, newRelease string) error {
+	const template = `%%changelog
+* %s Microsoft Golang Bot <microsoft-golang-bot@users.noreply.github.com> - %s-%s
```

Comment:
If this PR lands after https://github.com/microsoft/go-infra/pull/217, then we should change this line to the following:

```suggestion
* %s bot-for-go[bot] <199222863+bot-for-go[bot]@users.noreply.github.com> - %s-%s
```

Reply:
IMO including the username in the code at all is wrong (it should either be a console input or pulled from Git), and that magic number showing up has made it stand out more.